pkg/codegen/schema/bind.go

```diff
@@ -1049,6 +1049,37 @@ func bindDefaultValue(path string, value interface{}, spec *DefaultSpec, typ Typ
 	return dv, diags
 }
 
+func isOverlyNested(path string, typ Type) hcl.Diagnostics {
+	isCollection := func(typ Type) Type {
+		switch typ := typ.(type) {
```

Comment:
Another nit, feel free to ignore, at this point `typ` has been shadowed twice in the same function

Reply:
Very common for us to shadow in switch statements, and I don't think there's really a better name for the isCollection param, the shadowing is probably good there to ensure you don't reference the outer Type.